src/utils/filesystem.ts

```diff
@@ -0,0 +1,29 @@
+import { parse, resolve } from 'node:path'
+import { access, constants, mkdir, writeFile } from 'node:fs/promises'
+import { handleFileSystemError } from './error/filesystem-error'
+
+export const saveToFile = async (filePath: string, data: string) => {
+  // Check if the path exists, and create it if it doesn't
+  const resolvedPath = parse(filePath).dir
+  try {
+    await access(resolvedPath, constants.F_OK)
```

Comment:
this check is unnecessary, as the `mkdir` function with `{ recursive: true }` will succeed if the directory already exists so we don't need to check first